packages/react-query-next-experimental/src/HydrationStreamProvider.tsx

```diff
@@ -143,31 +143,28 @@ export function createHydrationStreamProvider<TShape>() {
       onEntriesRef.current = props.onEntries
     })
 
-    React.useEffect(() => {
+    // Setup and run the onEntries handler on the client only, but do it during
+    // the initial render so children have access to the data immediately
+    // This is important to avoid the client suspending during the initial render
+    // if the data has not yet been hydrated.
```

Comment:
Interesting callout! I _think_ it's a different situation here, and what we're doing is safe. `HydrationBoundary` could happen at any part of the tree, whereas there will only be one `ReactQueryStreamedHydration` and it will be high up in the tree / wrapping the whole thing. This code will only be called once on boot, and we went any streamed data to be available ASAP on the initial render phase.

Definitely let me know if you see any problems with that logic!

Reply:
Thanks for taking this across the finish line @TkDodo!